### PR TITLE
[Decomposition] hann_window.periodic

### PR DESCRIPTION
**Pull request by salilsdesai**
Summary: Add decomposition for hann_window.periodic and include it in core_aten_decompositions

Differential Revision: D48939996

### CODE REVIEW

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -4014,6 +4014,29 @@ def scaled_dot_product_flash_attention(
     )
 
 
+@register_decomposition(aten.hann_window.periodic)
+def hann_window_periodic(
```
Comment (larryliu0820) [inline torch/_decomp/decompositions.py:4018]: can we add some comment to explain the algorithm used here

**Review by SherlockNoMad — CHANGES_REQUESTED** [review]
I didn't see a test case in common_method_invocation... 
So we are missing test case in test_decomp...

**Comment by SherlockNoMad** [comment]
I didn't see a test case in common_method_invocation.py... 
So we are missing test case in test_decomp.py... 

> I didn't see a test case in common_method_invocation... So we are missing test case in test_decomp...

**Comment by salilsdesai** [comment]
> I didn't see a test case in common_method_invocation... So we are missing test case in test_decomp...

Ok I added the test case!

File: torch/_decomp/decompositions.py (hunk 1 above)
Comment (salilsdesai) [reply torch/_decomp/decompositions.py:4018]: The algorithm is explained well on the torch.hann_window page. I feel like anyone who is unsure about how this works should be able to easily find that page.

https://pytorch.org/docs/stable/generated/torch.hann_window.html

**Review by SherlockNoMad — APPROVED** [review]
LGTM, remember to export the test case change to OSS.

**Comment by salilsdesai** [comment]
PR isn't being updated anymore properly so i have to re-export the difff

File: torch/_decomp/decompositions.py [hunk 2]
```diff
@@ -4014,6 +4014,29 @@ def scaled_dot_product_flash_attention(
...
+    *,
+    dtype: Optional[torch.dtype] = None,
+    layout: Optional[torch.layout] = None,
+    device: Optional[torch.device] = None,
+    pin_memory: Optional[bool] = None,
+) -> Tensor:
+    # aten::hann_window.periodic(int window_length, bool periodic, ...) -> Tensor
+    N = window_length + (1 if periodic else 0)
+    return torch.narrow(
+        torch.pow(torch.sin((torch.arange(0, N) * torch.pi) / (N - 1)), 2),
```
[the remark this answers is not in the document]
Comment (salilsdesai) [reply torch/_decomp/decompositions.py:4030]: This is an old PR, please see this other one
https://github.com/pytorch/pytorch/pull/109464

**Pull request closed without merging**